client/knowledge_base_search/src/App.js

```diff
@@ -5,24 +5,44 @@ import { Header, Jumbotron, Footer, Icon } from 'watson-react-components';
 import SearchContainer from './containers/SearchContainer/SearchContainer';
 import ResultsContainer from './containers/ResultsContainer/ResultsContainer';
 import ErrorContainer from './containers/ErrorContainer/ErrorContainer';
+import ViewAllContainer from './containers/ViewAllContainer/ViewAllContainer';
 import links from './utils/links';
 import query from './actions/query';
+import questions from './actions/questions';
 import 'watson-react-components/dist/css/watson-react-components.css';
 import './App.css';
 
 class App extends Component {
-  componentWillMount() {
+  constructor(props) {
+    super(props);
     this.state = {
-      fetching: false,
-      results_fetched: false,
+      fetchingQuestions: true,
+      fetchingResults: false,
+      resultsFetched: false,
       results: [],
       enriched_results: [],
       search_input: '',
       results_error: null,
-      searchContainerHeight: 0
+      questionsError: null,
+      searchContainerHeight: 0,
+      showViewAll: false,
+      presetQueries: []
     }
   }
 
+  componentDidMount() {
+    questions().then((response) => {
+      this.setState({
+        fetchingQuestions: false
+      });
+      if (response.error) {
+        this.setState({ questionsError: response.error });
+      } else {
+        this.setState({ presetQueries: this.shuffleQuestions(response) });
+      }
+    });
```

Comment:
A couple things to maybe think about. React batches multiple set states so long as you're within a life cycle or handled snythetic event (e.g. `onClick`, `onSubmit`). It does not handle batching within async functions so I assume we're not getting batched here. Maybe using `await` would fix that?

If `await` isn't possible here, I'm not sure of the best way to get this reduced to one `this.setState` to prevent multiple re-renders. I guess the first thing to check would be to see if it's even re-rendering multiple times. Maybe batching works with promises so long as they're within a life cycle event?